plugins/inputs/nsq/nsq.go

```diff
@@ -78,21 +110,54 @@ func (n *NSQ) Gather(acc telegraf.Accumulator) error {
 	return nil
 }
 
-var tr = &http.Transport{
-	ResponseHeaderTimeout: time.Duration(3 * time.Second),
+func (n *NSQ) buildTLSConfig() (*tls.Config, error) {
+	if n.TlsCert == "" || n.TlsKey == "" || n.TlsCacert == "" {
+		return nil, nil
+	}
+
+	caCertBytes, err := ioutil.ReadFile(n.TlsCacert)
+	if err != nil {
+		return nil, fmt.Errorf("fail to read CA cert file %v: %v", n.TlsCacert, err)
+	}
+
+	cert, err := tls.LoadX509KeyPair(n.TlsCert, n.TlsKey)
+	if err != nil {
+		return nil, fmt.Errorf("fail to load certificate %v: %v", n.TlsCert, err)
+	}
+
+	pool := x509.NewCertPool()
+	pool.AppendCertsFromPEM(caCertBytes)
+
+	config := &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		ClientCAs:    pool,
+		RootCAs:      pool,
+	}
+
+	config.Rand = rand.Reader
+	return config, nil
 }
 
-var client = &http.Client{
-	Transport: tr,
-	Timeout:   time.Duration(4 * time.Second),
+func (n *NSQ) getHttpClient() *http.Client {
+	n.httpClientOnce.Do(func() {
+		tr := &http.Transport{
+			ResponseHeaderTimeout: time.Duration(3 * time.Second),
```

Comment:
Could you remove the ResponseHeaderTimeout and only use the Timeout on http.Client?